lib/ui/painting/image_decoder_no_gl_unittests.cc

```diff
@@ -0,0 +1,212 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/lib/ui/painting/image_decoder_no_gl_unittests.h"
+
+namespace flutter {
+namespace testing {
+
+// Tests are disabled for fuchsia.
+#if defined(OS_FUCHSIA)
+#pragma GCC diagnostic ignored "-Wunreachable-code"
+#endif
+
+namespace {
+
+bool IsPngWithPLTE(const uint8_t* bytes, size_t size) {
+  if (size < 8) {
+    return false;
+  }
+
+  if (memcmp(bytes, "\x89PNG\x0d\x0a\x1a\x0a", 8) != 0) {
+    return false;
+  }
+
+  const uint8_t* end = bytes + size;
+  const uint8_t* loc = bytes + 8;
+  while (loc + 8 <= end) {
+    uint32_t chunk_length =
+        (loc[0] << 24) | (loc[1] << 16) | (loc[2] << 8) | loc[3];
+
+    if (memcmp(loc + 4, "PLTE", 4) == 0) {
+      return true;
+    }
+
+    loc += /*length*/ 4 + /*type*/ 4 + chunk_length + /*crc*/ 4;
+  }
+
+  return false;
+}
+
+sk_sp<SkData> OpenFixtureAsSkData(const char* name) {
+  auto fixtures_directory =
+      fml::OpenDirectory(GetFixturesPath(), false, fml::FilePermission::kRead);
+  if (!fixtures_directory.is_valid()) {
+    return nullptr;
+  }
+
+  auto fixture_mapping =
+      fml::FileMapping::CreateReadOnly(fixtures_directory, name);
+
+  if (!fixture_mapping) {
+    return nullptr;
+  }
+
+  SkData::ReleaseProc on_release = [](const void* ptr, void* context) -> void {
+    delete reinterpret_cast<fml::FileMapping*>(context);
+  };
+
+  auto data = SkData::MakeWithProc(fixture_mapping->GetMapping(),
+                                   fixture_mapping->GetSize(), on_release,
+                                   fixture_mapping.get());
+
+  if (!data) {
+    return nullptr;
+  }
+  // The data is now owned by Skia.
+  fixture_mapping.release();
+  return data;
+}
+
+}  // namespace
+
+float HalfToFloat(uint16_t half) {
+  switch (half) {
+    case 0x7c00:
+      return std::numeric_limits<float>::infinity();
+    case 0xfc00:
+      return -std::numeric_limits<float>::infinity();
+  }
+  bool negative = half >> 15;
+  uint16_t exponent = (half >> 10) & 0x1f;
+  uint16_t fraction = half & 0x3ff;
+  float fExponent = exponent - 15.0f;
+  float fFraction = static_cast<float>(fraction) / 1024.f;
+  float pow_value = powf(2.0f, fExponent);
+  return (negative ? -1.f : 1.f) * pow_value * (1.0f + fFraction);
+}
+
+float DecodeBGR10(uint32_t x) {
+  const float max = 1.25098f;
+  const float min = -0.752941f;
+  const float intercept = min;
+  const float slope = (max - min) / 1024.0f;
+  return (x * slope) + intercept;
+}
+
+TEST(ImageDecoderNoGLTest, ImpellerWideGamutDisplayP3) {
```

Comment:
This test was moved to its own file because it wasn't running on macos because other tests in image_decoder_unittests.cc need opengl, not this one.